src/cargo/ops/cargo_install.rs

```diff
@@ -768,13 +781,36 @@ pub fn uninstall_one(
     config: &Config,
 ) -> CargoResult<()> {
     let crate_metadata = metadata(config, root)?;
-    let mut metadata = read_crate_list(&crate_metadata)?;
+    let metadata = read_crate_list(&crate_metadata)?;
+    let pkgid = PackageIdSpec::query_str(spec, metadata.v1.keys())?.clone();
+    uninstall_one_impl(crate_metadata, metadata, &pkgid, bins, config)
+}
+
+fn uninstall_cwd(
+    root: &Filesystem,
+    bins: &[String],
+    config: &Config,
+) -> CargoResult<()> {
+    let crate_metadata = metadata(config, root)?;
+    let metadata = read_crate_list(&crate_metadata)?;
+    let source_id = SourceId::for_path(config.cwd())?;
+    let (pkg, _source) = select_path_pkg(None, &source_id, None, config, true)?;
+    let pkgid = pkg.package_id();
+    uninstall_one_impl(crate_metadata, metadata, pkgid, bins, config)
+}
+
+fn uninstall_one_impl(
+    crate_metadata: FileLock,
+    mut metadata: CrateListingV1,
+    pkgid: &PackageId,
+    bins: &[String],
+    config: &Config,
+) -> CargoResult<()> {
     let mut to_remove = Vec::new();
     {
-        let result = PackageIdSpec::query_str(spec, metadata.v1.keys())?.clone();
-        let mut installed = match metadata.v1.entry(result.clone()) {
+        let mut installed = match metadata.v1.entry(pkgid.clone()) {
             Entry::Occupied(e) => e,
-            Entry::Vacant(..) => panic!("entry not found: {}", result),
+            Entry::Vacant(..) => panic!("entry not found: {}", pkgid),
```

Comment:
This panic happens if you attempt to `cargo uninstall` something that is not installed.

Reply:
This is the status quo despite this change. What change are you requesting?